substrate/client/network/src/discovery.rs

```diff
@@ -214,23 +214,14 @@ impl DiscoveryConfig {
 			enable_mdns,
 			kademlia_disjoint_query_paths,
 			kademlia_protocol,
-			kademlia_legacy_protocol,
+			kademlia_legacy_protocol: _,
 			kademlia_replication_factor,
 		} = self;
 
 		let kademlia = if let Some(ref kademlia_protocol) = kademlia_protocol {
-			let mut config = KademliaConfig::default();
+			let mut config = KademliaConfig::new(kademlia_protocol.clone());
 
 			config.set_replication_factor(kademlia_replication_factor);
-			// Populate kad with both the legacy and the new protocol names.
-			// Remove the legacy protocol:
-			// https://github.com/paritytech/polkadot-sdk/issues/504
```

Comment:
I think we should wait a bit and scrape the network. I believe everything should be fine, but let's double check first.

I guess we can merge https://github.com/paritytech/polkadot-sdk/pull/6109 before this one as it also handles legacy_protocol removal, but shouldn't matter too much.

I'll have a look this week and investigate versions, generally I think its safe for now

Reply:
Yes, I specifically used `kademlia_legacy_protocol: _,` above instead of full removal of the property to reduce the inevitable conflict with https://github.com/paritytech/polkadot-sdk/pull/6109 whichever ends up being first.